### ISSUE TITLE
My Template is not really useful and user friendly.

### ISSUE THREAD

**Issue by pulla** [pojome/elementor#4166]
First of all, I am a huge fan of Elementor.
I love it and using a lot. But getting stress for "My template". If you have few, it's ok. But if you use a lot, it's really messy and hard to work.

**1. I can't update (overwrite )**
It's not easy to update current template I created earlier.
ex) my-block-home and I have to delete the previous one and add new one whenever I update.

**2. Hard to find.**
Hard to see how it looks like. It needs a preview (or thumbnail) to easy to find.

**3. Hard to sort**
If we can have some categories to sort my templates, it will be much better.
My template lists are really messy now. There are too many templates I created. Because I love Elementor and want to build my styles. But it's really hard to organize.

I am using Elementor everyday and keep working on building pages. 
The most problem is that I can't keep updating the same templates.
Whenever I save my work in "My template", I feel like it's not really user friendly. 
I have to delete the old one and add new one.  I started add version ( ex. my-own-template-v1.0 ) to figure out what I am working. Then it's really messy now.

**More use, more messy and frustrating.** I don't think you mean it to feel so.

I really hope you can add some more features in "My template". It might be good for theme authors to create some predesigned templates.
**Maybe you can make like Elementor "Pages", "Blocks".**

Thank you,

**Comment by dasigna** [comment]
+1
preview and sorting after sections and pages would be helpful already.
and seeing the list of existing templates for better naming control as well.

**Comment by amaru7** [comment]
I also agree and have the same issue with my templates. Having to open preview for every single template is really impractical if there are plenty of templates installed.

Also installing many templates really slows down Elementor too much. As much as I like this part, I dislike its lack of preview and slowness.

Hope it gets improved soon.

**Comment by bainternet** [comment]
Just to be clear, you are talking about the admin or the modal in the editor?

and what are your suggestions?

**Comment by pulla** [comment]
@bainternet The modal in the editor (preview). That's what most of people use.

We really want to have overwriteoption.
Let's say I created "home1" and keep working. when we want to update (overwrite) it, it's not possible.
We have to remove the old one and create a new one. 

It's ok, when you have only 1 or few. But if you have more than 10 (which are most of people will have ), then it's really complicated and headache.

Suggestion 1 : there should be overwrite (update or synchronize ) to current template. We don't want to remove and create a new one.

![image](https://user-images.githubusercontent.com/5612050/39505663-d00782c8-4e0e-11e8-9c2d-ec82df1e2d0b.png)

And hard to find, sort as well.
We hope if there is category for sort and my favorite buttons. it's just like yours.

![image](https://user-images.githubusercontent.com/5612050/39505699-0faf2688-4e0f-11e8-86c2-7be3ae04f532.png)

I hope you consider this feature. "overwrite" is mostly needed.

**Comment by davidmabelle** [comment]
Completely agree with above comments.  PLEASE show a thumnail of each template and a hover over to show a larger preview of the template.  Importing templates takes too long as well and sometimes does not accurately replicate the original template.   I wish there were an easy way to vote for features.  I do not think GitHub is the best place to capture these feature requests and bugs.

**Comment by pulla** [comment]
YES  Please. I love elementor and play (work) everyday.  That means more and more templates now. It's so messy. I can't remember which ones. I named them each. so it's many and complicated. More love elementor, more headche in my templates. But still love elementor ;)  I don't use any others.

**Comment by cavalierlife** [comment]
There is a plugin that lets you categorize your templates and filter by those categories. It's called Builder Template Categories - https://wordpress.org/plugins/builder-template-categories/

**Comment by dbjpanda** [comment]
Waiting for this feature since 1 year. :(  Any progress ?

**Comment by cavalierlife** [comment]
Yes, there has been progress. Version 2.4 introduced template categories. https://docs.elementor.com/article/492-elementor-2-4-overview

**Comment by gwmbox** [comment]
+1 for much of the above.

The ability to add an image preview for each template is a must to help the user see what they need.  It shoudl appear like the template library with all the templates shown visually.  If My Templates could be like that it would be far easier to use.  Sorting and overwrite are a must too.

**Comment by sproutwp** [comment]
Guys.

Please. It's been 2 years since I made the first request for what is literally the most basic thing to have in a page builder. This feature alone is literally why we've used other builders over Elementor alone.

We'd switch in a heart-beat.

**Comment by tdechangy** [comment]
categories is on the good way but >>> PREVIEW, PREVIEW, PREVIEW, add a permanent thumbnail PREVIEW on the templates page and template loading popup pls. That will make our working process effective, and not looking elsewhere. Not to name it, Oxygen builder looks really smooth in this process, even more in global color management.
Million users are waiting for such basic enhancements.

**Comment by hipnetic** [comment]
I'm a new Elementor Pro user and would like to add my vote for this.  Brizy offers a thumbnail view (but is otherwise worse than Elementor's support).

Also, I don't think I've seen anyone ask for this, but I'd also like to be able to save stacked sections as a single block/template.  For example, there are some Elementor Pro Blocks (e.g., some headers) that are actually stacked sections, but it doesn't look like I can save stacked sections of my own as a template.  Instead, I'd have to save each part of the stacked section as a separate template.  Or am I missing something?

**Comment by damienrave** [comment]
+1 on the preview thing. When designing reusable blocks for clients with no technical/design background, naming templates in a meaningful way can be a real pain (TWO-COLUMN-WITH-LARGE-PHOTO vs THREE-COLUMN-WHITE-SQUARE-BLOCKS-WITH-ICON anyone?). We really would benefit from a thumbnail preview of custom templates... Thanks in advance !

**Comment by black-eye** [comment]
+1

**Comment by kingfisher64** [comment]
Any update on this? I see pages/blocks tab both have nice previews but the "my templates" doesn't. https://analogwp.com/custom-library-for-elementor/ plugin is basically a fix for these things but they should be in core as it's simply UX improvement.
